src/Interfaces/hamiltonians.jl

```diff
@@ -419,3 +432,35 @@ See also [`LOStructure`](@ref Main.Hamiltonians.LOStructure).
 has_adjoint(op) = has_adjoint(LOStructure(op))
 has_adjoint(::AdjointUnknown) = false
 has_adjoint(::LOStructure) = true
+
+"""
+    OperatorColumn(op::AbstractOperator{T}, address::A) <: AbstractVector{Pair{A,T}}
+
+Default column, using offdiagonals and diagonal_element.
+"""
+struct OperatorColumn{A,T,O<:AbstractOperator{T}} <: AbstractVector{Pair{A,T}}
+    operator::O
+    address::A
+    length::Int
+    diagonal::T
+end
```

Comment:
```suggestion
struct OperatorColumnFromOffdiagonals{A,T,O<:AbstractOffdiagonals{A,T}} <: AbstractVector{Pair{A,T}}
    offdiagonals::O
    diagonal::T
end
```
When defining a default `operator_column` that is based on `offdiagonals` it would make more sense to store the struct created by `offdiagonals`. This way any overheads that go towards creating the `offdiagonals` only has to be paid once. The length, address type, and value type can all be taken from the type instance. `getindex` would just index into the offdiagonals instance.

Reply:
I would leave it like this and turn all `offdiagonals` structs we have into columns.